csi/controller.go

```diff
@@ -315,11 +316,154 @@ func osdVolumeAttributes(v *api.Volume) map[string]string {
 	}
 }
 
+// CreateVolume is a CSI API which creates a volume on OSD
+// This function supports snapshots if the parent volume id is supplied
+// in the parameters.
+func (s *OsdCsiServer) CreateVolume(
+	ctx context.Context,
+	req *csi.CreateVolumeRequest,
+) (*csi.CreateVolumeResponse, error) {
+
+	// Log request
```

Comment:
Add nil check for req ?

Reply:
afaik, grpc will not call with nil, it will not allow it.